selfservice/flow/error.go

```diff
@@ -27,6 +27,42 @@ var (
 	ErrStrategyAsksToReturnToUI = errors.New("flow strategy is redirecting to the ui")
 )
 
+// Is sent when a flow is replaced by a different flow of the same class
+//
+// swagger:model selfServiceFlowReplacedError
+type ReplacedError struct {
+	*herodot.DefaultError `json:"error"`
+
+	// The flow ID that should be used for the new flow as it contains the correct messages.
+	FlowID uuid.UUID `json:"use_flow_id"`
+
+	flow Flow
+
+	// TODO: This error could be enhanced by providing a "flow class" (e.g. "Recovery", "Settings", "Verification", "Login", etc.)
```

Comment:
We could use this error type to describe "flow redirects" as well, since flow IDs are always `UUID`. 

We also might consider using this in place of the very specific `ExpiredError` (below). The `ExpiredError`s fields (`ExpiredAt`, etc.) could be added via a generic JSON document named `context` or similar. That way, we have a generic error for these cases. But that needs more design & thought, IMO.

Reply:
yes, I use it if the flow needs to be retried for other reasons than being expired (e.g. CSRF violations, or DB errors, etc.)